pkg/ccl/backupccl/restore_planning.go

```diff
@@ -1584,6 +1584,15 @@ func doRestorePlan(
 		return err
 	}
 
+	currentVersion := p.ExecCfg().Settings.Version.ActiveVersion(ctx)
```

Comment:
Perhaps a comment here mentioning that cluster versions are not expected to ever decrease even in the case of rolling back to an earlier patch version would be useful.